op-acceptance-tests/tests/fjord/check_scripts_test.go

```diff
@@ -1,89 +1,240 @@
-		checkFjordConfig := &fjordChecks.CheckFjordConfig{
-			Log:  logger,
-			L2:   l2LowLevelClient,
-			Key:  privateKey,
-			Addr: walletAddr,
-		}
+		if len(tc.data) == 0 {
+			plannedTx := wallet.Transfer(walletAddr, eth.ZeroWei)
+			var err error
+			receipt, err = plannedTx.Included.Eval(ctx)
+			require.NoError(err)
+			require.NotNil(receipt)
+
+			_, txs, err := l2Client.InfoAndTxsByHash(ctx, receipt.BlockHash)
+			require.NoError(err)
 
-		block, err := l2.Nodes()[0].BlockByNumber(t.Context(), nil)
-		require.NoError(t, err)
-		time := block.Time()
-
-		isFjordActivated, err := validators.IsForkActivated(chainConfig, rollup.Fjord, time)
-		require.NoError(t, err)
-
-		if !isFjordActivated {
-			err = fjordChecks.CheckRIP7212(t.Context(), checkFjordConfig)
-			require.Error(t, err, "expected error for CheckRIP7212")
-			err = fjordChecks.CheckGasPriceOracle(t.Context(), checkFjordConfig)
-			require.Error(t, err, "expected error for CheckGasPriceOracle")
-			err = fjordChecks.CheckTxEmpty(t.Context(), checkFjordConfig)
-			require.Error(t, err, "expected error for CheckTxEmpty")
-			err = fjordChecks.CheckTxAllZero(t.Context(), checkFjordConfig)
-			require.Error(t, err, "expected error for CheckTxAllZero")
-			err = fjordChecks.CheckTxAll42(t.Context(), checkFjordConfig)
-			require.Error(t, err, "expected error for CheckTxAll42")
-			err = fjordChecks.CheckTxRandom(t.Context(), checkFjordConfig)
-			require.Error(t, err, "expected error for CheckTxRandom")
+			for _, tx := range txs {
+				if tx.Hash() == receipt.TxHash {
+					signedTx = tx
+					break
+				}
+			}
+			require.NotNil(signedTx)
 		} else {
-			err = fjordChecks.CheckAll(t.Context(), checkFjordConfig)
-			require.NoError(t, err, "should not error on CheckAll")
+			opt := txplan.Combine(
+				wallet.Plan(),
+				txplan.WithTo(&walletAddr),
+				txplan.WithValue(eth.ZeroWei.ToBig()),
+				txplan.WithData(tc.data),
+			)
+			plannedTx := txplan.NewPlannedTx(opt)
+			var err error
+			receipt, err = plannedTx.Included.Eval(ctx)
+			require.NoError(err)
+			require.NotNil(receipt)
+
+			_, txs, err := l2Client.InfoAndTxsByHash(ctx, receipt.BlockHash)
+			require.NoError(err)
+
+			for _, tx := range txs {
+				if tx.Hash() == receipt.TxHash {
+					signedTx = tx
+					break
+				}
+			}
+			require.NotNil(signedTx)
 		}
+
+		require.Equal(uint64(1), receipt.Status)
+
+		blockNumber := receipt.BlockNumber
+		opts := &bind.CallOpts{Context: ctx, BlockNumber: blockNumber}
+
+		unsignedTx := types.NewTx(&types.DynamicFeeTx{
+			Nonce:     signedTx.Nonce(),
+			To:        signedTx.To(),
+			Value:     signedTx.Value(),
+			Gas:       signedTx.Gas(),
+			GasFeeCap: signedTx.GasFeeCap(),
+			GasTipCap: signedTx.GasTipCap(),
+			Data:      signedTx.Data(),
+		})
+
+		txUnsigned, err := unsignedTx.MarshalBinary()
+		require.NoError(err)
+		txSigned, err := signedTx.MarshalBinary()
+		require.NoError(err)
+
+		gpoL1GasUsed, err := gasPriceOracle.GetL1GasUsed(opts, txUnsigned)
```

Comment:
nit; `gpoL1GasUsed` seems unused?

```suggestion
		_, err := gasPriceOracle.GetL1GasUsed(opts, txUnsigned)
```